### PR TITLE
Lookup: add Vue and React code snippets

### PR DESCRIPTION
**Pull request by ksercs**

### CODE REVIEW

File: concepts/05 Widgets/Lookup/10 Enable Grouping.md [hunk 1]
```diff
@@ -108,21 +180,95 @@ If you use the [DevExtreme DataSource](/api-reference/30%20Data%20Layer/DataSour
...
+                />
+            );
+        }
+    }
+
+    export default App;
 
 ---
 
 You can define a custom template for the group headers. For Angular, AngularJS and Knockout apps, DevExtreme provides the [dxTemplate](/api-reference/10%20UI%20Widgets/Markup%20Components/dxTemplate '/Documentation/ApiReference/UI_Widgets/Markup_Components/dxTemplate/') markup component. The following code shows how to use **dxTemplate** to define a template for the **Lookup** group headers.
```
Comment (ksercs) [inline concepts/05 Widgets/Lookup/10 Enable Grouping.md]: Please, help me to rework this paragraph

File: concepts/05 Widgets/Lookup/03 Data Binding/05 Simple Array/05 Array Only.md [hunk 2]
```diff
@@ -129,12 +242,82 @@ You can create a [Query](/concepts/70%20Data%20Binding/5%20Data%20Layer/6%20Quer
...
+
+    import React from 'react';
+    import 'devextreme/dist/css/dx.common.css';
+    import 'devextreme/dist/css/dx.light.css';
+
+    import { Lookup } from 'devextreme-react/lookup';
+    import query from "devextreme/data/query";
+
+    class App extends React.Component {
+        constructor(props) {
```
Comment (RomanTsukanov) [inline concepts/05 Widgets/Lookup/03 Data Binding/05 Simple Array/05 Array Only.md:294]:
```suggestion
        constructor(props) {
            super(props);
```

File: concepts/05 Widgets/Lookup/03 Data Binding/05 Simple Array/10 ArrayStore.md [hunk 3]
```diff
@@ -39,22 +46,85 @@ Extend a JavaScript array's functionality by placing it into an [ArrayStore](/ap
...
+    import 'devextreme/dist/css/dx.common.css';
+    import 'devextreme/dist/css/dx.light.css';
+
+    import { Lookup } from 'devextreme-react/lookup';
+    import ArrayStore from "devextreme/data/array_store";
+
+    const products = [/* ... */ ];
+
+    class App extends React.Component {
+        constructor(props) {
```
Comment (RomanTsukanov) [inline concepts/05 Widgets/Lookup/03 Data Binding/05 Simple Array/10 ArrayStore.md:97]:
```suggestion
        constructor(props) {
            super(props);
```

Comment (RomanTsukanov) [inline concepts/05 Widgets/Lookup/03 Data Binding/05 Simple Array/10 ArrayStore.md:97]: Please check this in other examples.

File: concepts/05 Widgets/Lookup/10 Enable Grouping.md [hunk 4]
```diff
@@ -43,11 +49,69 @@ The **Lookup** can organize items in groups. If you use a simple array as a data
...
+    import { DxLookup } from 'devextreme-vue/lookup';
+
+    export default {
+        components: {
+            DxLookup
+        },
+        data() {
+            return {
+                lookupData: [{
+                        key: 'Mr. John Heart',
```
Comment (RomanTsukanov) [inline concepts/05 Widgets/Lookup/10 Enable Grouping.md]: Too much indentation.

File: concepts/05 Widgets/Lookup/10 Enable Grouping.md [hunk 5]
```diff
@@ -43,11 +49,69 @@ The **Lookup** can organize items in groups. If you use a simple array as a data
...
+##### React
+
+    import React from 'react';
+    import 'devextreme/dist/css/dx.common.css';
+    import 'devextreme/dist/css/dx.light.css';
+
+    import { Lookup } from 'devextreme-react/lookup';
+
+    const lookupData = [{
+            key: 'Mr. John Heart',
```
Comment (RomanTsukanov) [inline concepts/05 Widgets/Lookup/10 Enable Grouping.md]: Here too.

File: concepts/05 Widgets/Lookup/10 Enable Grouping.md (hunk 1 above)
Comment (RomanTsukanov) [reply concepts/05 Widgets/Lookup/10 Enable Grouping.md]:
```suggestion
To customize group headers, specify a [groupTemplate](/Documentation/ApiReference/UI_Widgets/dxLookup/Configuration/#groupTemplate). In Angular and Vue, you can declare it in the markup. In React, you can use a rendering function (shown in the code below) or component:
```

File: concepts/05 Widgets/Lookup/20 Customize the Appearance/05 Customize Item Appearance.md [hunk 6]
```diff
@@ -36,19 +43,72 @@ For a minor customization of **Lookup** items, you can define [specific fields](
...
+            );
+        }
+    }
+
+    export default App;
 
 ---
 
-If you need a more flexible solution, define a custom template for widget items. For Angular, AngularJS and Knockout apps, DevExtreme provides the [dxTemplate](/api-reference/10%20UI%20Widgets/Markup%20Components/dxTemplate '/Documentation/ApiReference/UI_Widgets/Markup_Components/dxTemplate/') markup component. The following code shows how to use **dxTemplate** to define a template for the **Lookup** items.
+If you need a more flexible solution, define an [itemTemplate](/api-reference/10%20UI%20Widgets/CollectionWidget/1%20Configuration/itemTemplate.md '/Documentation/ApiReference/UI_Widgets/dxTabs/Configuration/#itemTemplate'). In Angular and Vue, you can declare it in the markup. In React, you can use a rendering function (shown in the code below) or component:.
```
Comment (RomanTsukanov) [inline concepts/05 Widgets/Lookup/20 Customize the Appearance/05 Customize Item Appearance.md]:
```suggestion
If you need a more flexible solution, define an [itemTemplate](/api-reference/10%20UI%20Widgets/CollectionWidget/1%20Configuration/itemTemplate.md '/Documentation/ApiReference/UI_Widgets/dxLookup/Configuration/#itemTemplate'). In Angular and Vue, you can declare it in the markup. In React, you can use a rendering function (shown in the code below) or component:.
```

**Review by RomanTsukanov — CHANGES_REQUESTED** [review]

File: concepts/05 Widgets/Lookup/10 Enable Grouping.md (hunk 1 above)
Comment (ksercs) [reply concepts/05 Widgets/Lookup/10 Enable Grouping.md]: fixed

File: concepts/05 Widgets/Lookup/03 Data Binding/05 Simple Array/05 Array Only.md (hunk 2 above)
Comment (ksercs) [reply concepts/05 Widgets/Lookup/03 Data Binding/05 Simple Array/05 Array Only.md:294]: fixed

File: concepts/05 Widgets/Lookup/03 Data Binding/05 Simple Array/10 ArrayStore.md (hunk 3 above)
Comment (ksercs) [reply concepts/05 Widgets/Lookup/03 Data Binding/05 Simple Array/10 ArrayStore.md:97]: fixed

Comment (ksercs) [reply concepts/05 Widgets/Lookup/03 Data Binding/05 Simple Array/10 ArrayStore.md:97]: fixed

File: concepts/05 Widgets/Lookup/10 Enable Grouping.md (hunk 4 above)
Comment (ksercs) [reply concepts/05 Widgets/Lookup/10 Enable Grouping.md]: fixed

File: concepts/05 Widgets/Lookup/10 Enable Grouping.md (hunk 5 above)
Comment (ksercs) [reply concepts/05 Widgets/Lookup/10 Enable Grouping.md]: fixed

File: concepts/05 Widgets/Lookup/20 Customize the Appearance/05 Customize Item Appearance.md (hunk 6 above)
Comment (ksercs) [reply concepts/05 Widgets/Lookup/20 Customize the Appearance/05 Customize Item Appearance.md]: fixed

File: concepts/05 Widgets/Lookup/03 Data Binding/05 Simple Array/05 Array Only.md [hunk 7]
```diff
@@ -129,12 +242,84 @@ You can create a [Query](/concepts/70%20Data%20Binding/5%20Data%20Layer/6%20Quer
...
+        getSortedProducts() {
+            return query(this.products)
+                .sortBy("price", true)
+                .toArray();
+        }
+
+        render() {
+            return (
+                <Lookup
+                    dataSource={this.getSortedProducts()}
```
Comment (AlexJameson) [inline concepts/05 Widgets/Lookup/03 Data Binding/05 Simple Array/05 Array Only.md:314]: Shouldn't it be dataSource={this.getSortedProducts}?

Am I missing something?

**Review by AlexJameson — APPROVED** [review]

File: concepts/05 Widgets/Lookup/03 Data Binding/05 Simple Array/05 Array Only.md (hunk 7 above)
Comment (ksercs) [reply concepts/05 Widgets/Lookup/03 Data Binding/05 Simple Array/05 Array Only.md:314]: No, data source is the result of `getSortedProducts` function call

**Review by RomanTsukanov — APPROVED** [review]

**Pull request merged**